dds/DCPS/DiscoveryBase.h

```diff
@@ -1216,10 +1220,6 @@ namespace OpenDDS {
           return; // Possible and ok, since lock is released
         }
 
-        if (!writer_local && !reader_local) {
-          return;
-        }
-
```

Comment:
Should this be removed? When there is at least one local endpoint on this topic, should the function continue past this point when both sides are non-local?

Reply:
Then what about restoring my previous change in `match_endpoints` where we return before checking non-locals?

Reply:
Any change that helps scalability and doesn't hurt correctness is fair game